File: Backend/DTOs/AcademicPerformanceDTO.cs
namespace Backend.DTOs;

public class AcademicPerformanceDTO
{
    public List<SubjectInfolDTO> Subjects { get; set; } = default!;
    public int TotalStudents { get; set; }
    public int TotalStudentsWithScoresFrom0to40 { get; set; }
    public int TotalStudentsWithScoresFrom41to60 { get; set; }
    public int TotalStudentsWithScoresFrom61to90 { get; set; }
    public int TotalStudentsWithScoresFrom91to100 { get; set; }
}

File: frontend/src/app/dto/academic-performance.dto.ts
import {SubjectInfoDto} from "./subject-info.dto";

export interface AcademicPerformanceDto {
  subjects: SubjectInfoDto[];
  totalStudents: number;
  totalStudentsWithScoresFrom0to40: number;
  totalStudentsWithScoresFrom41to60: number;
  totalStudentsWithScoresFrom61to90: number;
  totalStudentsWithScoresFrom91to100: number;
}
